ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm_step2/device/moreh_clip_grad_norm_step2_program_factory.cpp

```diff
@@ -0,0 +1,163 @@
+    //                      Device Setup
+    ////////////////////////////////////////////////////////////////////////////
+    auto device = tmp_pow_sum.device();
+    auto program = CreateProgram();
+
+    ////////////////////////////////////////////////////////////////////////////
+    //                         Parameters Setup
+    ////////////////////////////////////////////////////////////////////////////
+    const auto num_tiles = tmp_pow_sum.volume() / tt::constants::TILE_HW;
+
+    auto [p, decimal, p_is_negative] = get_p_decimal_p_is_negative(1.0f / norm_type);
+
+    ////////////////////////////////////////////////////////////////////////////
+    //                         Core Setup
+    ////////////////////////////////////////////////////////////////////////////
+    CoreCoord single_core = {0, 0};
+
+    ////////////////////////////////////////////////////////////////////////////
+    //                         CircularBuffer Setup
+    ////////////////////////////////////////////////////////////////////////////
+    const uint32_t in0_t = 1;  // input(==tmp_pow_sum)
+    const uint32_t in1_t = 1;  // decimal
+
+    // x^p * exp(log(x) * decimal)
+    const uint32_t out0_t = 1;  // output(==total_norm)
+
+    const uint32_t im0_t = 1;  // Sum[tmp_pow_sum](==x)
+    const uint32_t im1_t = 1;  // x^p
+    const uint32_t im2_t = 1;  // log(x)
+    const uint32_t im3_t = 1;  // exp(log(x) * decimal)
+
+    const auto cb_data_format = tt::tt_metal::datatype_to_dataformat_converter(total_norm.get_dtype());
+
+    ttnn::operations::CreateCircularBuffer(
+        program,
+        single_core,
+        cb_data_format,
+        {
+            {tt::CB::c_in0, in0_t},        // input(==tmp_pow_sum)
+            {tt::CB::c_in1, in1_t},        // decimal
+            {tt::CB::c_out0, out0_t},      // output(==total_norm)
+            {tt::CB::c_intermed0, im0_t},  // Sum[tmp_pow_sum](==x)
+            {tt::CB::c_intermed1, im1_t},  // x^p
+            {tt::CB::c_intermed2, im2_t},  // log(x)
+            {tt::CB::c_intermed3, im3_t},  // exp(log(x) * decimal)
+        });
+
+    ////////////////////////////////////////////////////////////////////////////
+    //                      DataMovementKernel SetUp
+    ////////////////////////////////////////////////////////////////////////////
+    const auto reader_kernel_file =
+        "ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm_step2/device/kernels/"
+        "reader_moreh_clip_grad_norm_step2.cpp";
+    const auto writer_kernel_file =
+        "ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm_step2/device/kernels/"
+        "writer_moreh_clip_grad_norm_step2.cpp";
+
+    const auto reader_kernel_id = ttnn::operations::CreateReadKernel(program, reader_kernel_file, single_core);
+    const auto writer_kernel_id = ttnn::operations::CreateWriteKernel(program, writer_kernel_file, single_core);
```

Comment:
```suggestion
    const auto reader_kernel_id = CreateReadKernel(program, reader_kernel_file, single_core);
    const auto writer_kernel_id = CreateWriteKernel(program, writer_kernel_file, single_core);
```

Reply:
Resolved in [06f7fe7](https://github.com/tenstorrent/tt-metal/pull/14667/commits/06f7fe78b8714d808e3442033a18a39ce88500c8)